packages/maps/src/components/MapView.tsx

```diff
@@ -216,10 +225,28 @@ const MapViewF = <T extends object>({
     }
   }, [latitude, longitude, zoom, animateToLocation]);
 
+  // Use delayed/debounced value to prevent too many calls when map is being dragged
+  React.useEffect(() => {
+    const callOnRegionChange = async () => {
+      if (delayedRegionValue) {
+        const camera = await mapRef.current?.getCamera();
+        onRegionChange?.({ ...delayedRegionValue, zoom: camera?.zoom || 1 });
+      }
+    };
+
+    callOnRegionChange();
+    // onRegionChange excluded to prevent calling on every rerender when using an anonymous function (which is most common)
```

Comment:
Ideally yes, but we don't do that in draftbit with actions, so that's why I went this direction. The exclusion of `onRegionChange` as a dependency would mean that if the value does not change, but the `onRegionChange` does, then the new callback won't get the current value, but will get all subsequent ones when the value changes. So it's not that big of a deal imo.

I think this is the better tradeoff of the two options. Either this or too many calls that could lead to endless loops.